v2/piv/piv.go

```diff
@@ -364,14 +369,39 @@ var (
 	aidYubiKey    = [...]byte{0xa0, 0x00, 0x00, 0x05, 0x27, 0x20, 0x01, 0x01}
 )
 
-func ykAuthenticate(tx *scTx, key [24]byte, rand io.Reader) error {
+func ykAuthenticate(tx *scTx, key []byte, rand io.Reader, version *version) error {
 	// https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-73-4.pdf#page=92
 	// https://tsapps.nist.gov/publication/get_pdf.cfm?pub_id=918402#page=114
 
+	var managementKeyType byte
+	if supportsVersion(version, 5, 3, 0) {
+		// if yubikey version >= 5.3.0, determine management key type using slot metadata
+		cmd := apdu{
+			instruction: insGetMetadata,
+			param1:      0x00,
+			param2:      keyCardManagement,
+		}
+		resp, err := tx.Transmit(cmd)
+		if err == nil {
```

Comment:
this should return an error, not ignore it

```
if err != nil {
    return fmt.Errorf("determining key management type: %v", err)
}
managementKeyType = resp[2:][0]
```

Reply:
Whoops, I missed this when I added the explicit version check rather than letting it attempt and silently fail to determine it. Fixed.